bot/utils/messages.py

```diff
@@ -238,3 +239,102 @@ async def send_denial(ctx: Context, reason: str) -> discord.Message:
 def format_user(user: discord.abc.User) -> str:
     """Return a string for `user` which has their mention and ID."""
     return f"{user.mention} (`{user.id}`)"
+
+
+def shorten_text(text: str) -> str:
+    """
+    Truncate the text if there are over 3 lines or 300 characters, or if it is a single word.
+
+    The maximum length of the string would be 303 characters across 3 lines at maximum.
+    """
+    original_length = len(text)
+    # Truncate text to a maximum of 300 characters
+    if len(text) > 300:
+        text = text[:300]
+
+    # Limit to a maximum of three lines
+    text = "\n".join(text.split("\n", maxsplit=3)[:3])
+
+    # If it is a single word, then truncate it to 50 characters
+    if text.find(" ") == -1:
+        text = text[:50]
+
+    # Remove extra whitespaces from the `text`
+    text = text.strip()
+
+    # Add placeholder if the text was shortened
+    if len(text) < original_length:
+        text = f"{text}..."
+
+    return text
+
+
+async def extract_message_links(message: discord.Message) -> Optional[list[str]]:
+    """This method extracts discord message links from a message object."""
+    return [msg_link[0] for msg_link in DISCORD_MESSAGE_LINK_RE.findall(message.content)]
```

Comment:
[0] here doesn't really make it clear what we're grabbing. We can remove this by fixing the message link re. See that other comment.